File: src/main/java/ninja/hudy/infosmog/repository/VoivodeshipRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.VoivodeshipRegion;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface VoivodeshipRepository extends JpaRepository<VoivodeshipRegion, Long> {

    Set<VoivodeshipRegion> findByName(String name);

    Optional<VoivodeshipRegion> findByCode(Integer code);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/VoivodeshipRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.VoivodeshipRegion
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

import java.util.Optional

@Repository
interface VoivodeshipRepository : JpaRepository<VoivodeshipRegion, Long> {
    fun findByName(name: String): Set<VoivodeshipRegion>

    fun findByCode(code: Int?): Optional<VoivodeshipRegion>
}
